Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard.py

```diff
@@ -0,0 +1,1203 @@
+        days_ago = first_fetch
+
+        fetch_days_ago = arg_to_datetime(arg=days_ago, arg_name="first_fetch", required=False)
+
+        demisto.debug(f"getLastRun is 'None' in Integration context, using parameter 'first_fetch' value '{fetch_days_ago}'")
+        demisto.debug(f"{days_ago} => {fetch_days_ago}")
+
+        return int(fetch_days_ago.timestamp())  # type: ignore
+
+
+def incidents_to_import(
+    alerts: List[Dict[str, Any]],
+    last_run: Optional[str] = demisto.getLastRun().get("last_run"),
+    first_fetch: Optional[str] = demisto.params().get("first_fetch", "2 days")
+) -> List[Dict[str, Any]]:
+    """
+    Helper function to filter events that need to be imported.
+    It filters the events based on the `created_at` timestamp.
+    Function will only be called if the SecurityScorecard API returns more than one alert.
+
+    Args:
+        ``alerts``(``List[Dict[str, Any]]``): A list of alerts to sort through.
+        ``last_run``(``str``): UNIX timestamp when the fetch incidents had last run.
+        ``first_fetch`` (``str``): Configured 'First fetch' parameter
+    Returns:
+        ``List[Dict[str, Any]]``: Alerts to import
+    """
+
+    last_run = get_last_run(last_run=last_run, first_fetch=first_fetch)  # type: ignore
+
+    demisto.debug(f"Last run timestamp: {last_run}")
+
+    incidents_to_import: List[Dict[str, Any]] = []
+
+    # Check if there are more than 0 alerts
+    if alerts:
+
+        # The alerts are sorted by ascending date so last alert is the most recent
+        most_recent_alert = alerts[-1]
+
+        most_recent_alert_created_date = most_recent_alert.get("created_at")
+
+        most_recent_alert_timestamp = \
+            int(datetime.strptime(most_recent_alert_created_date, SECURITYSCORECARD_DATE_FORMAT).timestamp())  # type: ignore
+        demisto.debug(f"Setting last runtime as alert most recent timestamp: {most_recent_alert_timestamp}")
+        demisto.setLastRun({
+            'last_run': most_recent_alert_timestamp
+        })
+
+        for alert in alerts:
+
+            alert_created_at = alert.get("created_at")
+            alert_timestamp = int(datetime.strptime(alert_created_at, SECURITYSCORECARD_DATE_FORMAT).timestamp())  # type: ignore
+
+            alert_id = alert.get("id")
+            company_name: str = alert.get("company_name")  # type: ignore
+            change_type: str = alert.get("change_type")  # type: ignore
+
+            debug_msg = f"import alert '{alert_id}'? (last_run < alert_timestamp): {(last_run < alert_timestamp)}"  # type: ignore
+
+            demisto.debug(debug_msg)
+
+            if alert_timestamp > last_run:  # type: ignore
+                incident = {}
+                incident["name"] = f"{company_name} {change_type.replace('_', ' ').title()}"
+                incident["occurred"] = \
+                    datetime.strptime(alert_created_at, SECURITYSCORECARD_DATE_FORMAT).strftime(DATE_FORMAT)  # type: ignore
```

Comment:
Use `arg_to_datetime` function here as well.

Reply:
Same as above:

I'm not using arg_to_datetime here because the alert date returned from the API is different than the one used by XSOAR:

SECURITYSCORECARD_DATE_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
arg_to_datetime doesn't have an argument for format so I could not use it.

Reply:
I think it should work on this format as well  (the only difference is with the milliseconds no?)
![content  ~:dev:demisto:content  -  :Packs:Carbon_Black_Enterprise_Response:Integrations:CarbonBlackResponseV2:CarbonBlackResp… 2021-10-10 12-42-37](https://user-images.githubusercontent.com/44666568/136690559-0aad605a-0ecd-496b-8792-ea6d9b669d4d.png)

Reply:
As well as the timezone information:

```
"alert_created_at: 2021-10-03T07:11:56.017Z"
"alert_datetime: 2021-10-03 07:11:56.017000+00:00"
"last_run: 2021-10-08 13:26:13.866124"
```

The timezone information was causing the following exception when attempting to compare the two:

```
TypeError: can't compare offset-naive and offset-aware datetimes
```

This is because `arg_to_datetime` was returns a `naive` `datetime` object (timezone unaware) while the `alert_datetime` is `aware`.

In any case, I made modifications to this in commit `9a682f4e7d23156c90b319a9c5555435cb93ddb5`

Reply:
I can't find commit `9a682f4e7d23156c90b319a9c5555435cb93ddb5` as part of this PR. 
In any case, last_run is something you have saved in the context (after taking it from the incident) so you should have the TZ, which means you can save it with the TZ when creating the last_run originally.

Reply:
fixed a couple of issues in commit 18c9d0093f:

- last alert was imported because it was larger in milliseconds than the last run (which doesn't include ms in date)
- fixed last run UTs where ms was provided:
```

E           assert datetime.datetime(2021, 10, 9, 22, 38, 2, 560000) == datetime.datetime(2021, 10, 9, 22, 38, 2)
E             +datetime.datetime(2021, 10, 9, 22, 38, 2, 560000)
E             -datetime.datetime(2021, 10, 9, 22, 38, 2)
```
- standardized all `datetime`s created by `arg_to_datetime` to use remove `tzinfo`, was causing error:
```
can't compare offset-naive and offset-aware datetimes
```